pkg/cri/resource-manager/control/cri/cri.go

```diff
@@ -86,7 +86,9 @@ func (ctl *crictl) PreCreateHook(c cache.Container) error {
 	create.Config.Envs = c.GetCRIEnvs()
 	create.Config.Mounts = c.GetCRIMounts()
 	create.Config.Devices = c.GetCRIDevices()
-	create.Config.Linux.Resources = c.GetLinuxResources()
+	if create.Config.Linux != nil {
+		create.Config.Linux.Resources = c.GetLinuxResources()
```

Comment:
While at it, it might be a good idea to check if there are other similar cases in our codebase (we get a complex object, which might be only partly initialized, back from a kubernetes API).